src/library/scala/concurrent/BatchingExecutor.scala

```diff
@@ -39,79 +40,76 @@ import scala.annotation.tailrec
  * WARNING: The underlying Executor's execute-method must not execute the submitted Runnable
  * in the calling thread synchronously. It must enqueue/handoff the Runnable.
  */
-private[concurrent] trait BatchingExecutor extends Executor {
+ trait BatchingExecutor extends Executor {
```

Comment:
What are the potential downsides?

Reply:
Usage?:)

Reply:
I'd keep it private since the implementation involves mutable state and people might try to extend it without understanding the implications. Advanced users can extend from `Executor` directly and implement something similar tailored to their needs.

Reply:
@fwbrasil True. Moved it back to private.